ops/charm.py

```diff
@@ -1176,6 +1180,29 @@ def add_status(self, status: model.StatusBase):
             model_.unit._collected_statuses.append(status)
 
 
+# FIXME: API design choice
+# Should we make this more generic?
+# - call this a TelemetryEvent / TelemeryConfigEvent / SetupTelemetryEvent
+# - provide .set_tracing_destination() in this PR
+# - later, perhaps .set_logging_xxx() and .set_metrics_yyy()?
+class SetupTracingEvent(LifecycleEvent):
+    """FIXME docstring."""
+
+    def set_destination(self, *, url: Optional[str], ca: Optional[str] = None) -> None:
```

Comment:
I've somewhat lost track of where things are documented, so this may be written already somewhere, sorry (feel free to just point me at it).

Did you try having something like `CharmBase.set_tracing_destination` instead of the event? It still feels weird to me to be setting up the tracing destination on every single hook, rather than once when I get the destination and then each time that relation data changes.